pkg/nsx/services/vpc/vpc.go

```diff
@@ -286,31 +288,91 @@ func (s *VPCService) getNetworkconfigNameFromNS(ns string) (string, error) {
 	return ncName, nil
 }
 
-func (service *VPCService) CreateorUpdateVPC(obj *v1alpha1.VPC) (*model.Vpc, error) {
+func (s *VPCService) GetDefaultSNATIP(vpc model.Vpc) (string, error) {
+	ruleClient := s.NSXClient.NATRuleClient
+	info, err := common.ParseVPCResourcePath(*vpc.Path)
+	if err != nil {
+		log.Error(err, "failed to parse vpc path to get default snat ip", vpc.Path)
+		return "", err
+	}
+	var cursor *string = nil
+	pageSize := int64(1000)
+	markedForDelete := false
+	results, err := ruleClient.List(info.OrgID, info.ProjectID, info.VPCID, common.DefaultSNATID, cursor, &markedForDelete, nil, &pageSize, nil, nil)
+	if err != nil {
+		log.Error(err, "failed to read snat rule list to get default snat ip", "VPC", vpc.Id)
+		return "", err
+	}
+
+	if results.Results == nil || len(results.Results) == 0 {
+		log.Info("no snat rule found under vpc", "VPC", vpc.Id)
+		return "", nil
+	}
+
+	// if there is multiple private ip block in vpc, there will also be multiple snat rules, but they are using
+	// the same snat ip, so just using the first snat rule to get snat ip.
+	return *results.Results[0].TranslatedNetwork, nil
+}
+
+func (s *VPCService) GetSubNetPathCidr(vpc model.Vpc) (string, string, error) {
+	subnetsClient := s.NSXClient.SubnetsClient
+	statusClient := s.NSXClient.SubnetStatusClient
+	info, err := common.ParseVPCResourcePath(*vpc.Path)
+	path := ""
+	cidr := ""
+	if err != nil {
+		log.Error(err, "failed to parse vpc path to get default snat ip", vpc.Path)
```

Comment:
as you return error, so no need to print error here? because the place calling this function would print the error again